### ISSUE TITLE
DataRequirementsProcessor throwing IllegalArgumentException when resolving let statement

### ISSUE THREAD

**Issue by jkotanchik-SB** [cqframework/clinical_quality_language#1147]
While running the DataRequirementsProcessor for Connectathon measure 645, the processor throws the following Exception when attempting to resolve the following `let`:
```
java.lang.IllegalArgumentException: Could not resolve let DoseTiming
```
The `DataRequiremetnsProcessor` is used as follows:
```
    // setting up the librarySourceProvider to fetch included libraries
    cqlConversionService.setUpLibrarySourceProvider(cql, accessToken);

    var translationResource = TranslationResource.getInstance(true);
    CqlTranslator cqlTranslator =
        translationResource.buildTranslator(
            new ByteArrayInputStream(cql.getBytes()), createDefaultRequestData().createMap());
    CompiledLibrary translatedLibrary = cqlTranslator.getTranslatedLibrary();
    LibraryManager libraryManager = translationResource.getLibraryManager();

    // providing compiled measureLibrary, as it cannot be fetched using LibrarySourceProvider ( we
    // are not storing measure libraries in HAPI)
    libraryManager.cacheLibrary(translatedLibrary);

    Set<String> expressionList = getExpressions(r5Measure);
    var dqReqTrans = new DataRequirementsProcessor();
    CqlTranslatorOptions options = CqlTranslatorOptions.defaultOptions();

    org.hl7.fhir.r5.model.Library effectiveDataRequirements =
        dqReqTrans.gatherDataRequirements(
            libraryManager, translatedLibrary, options, expressionList, true);
```
I noted when the processor attempted to resolve the let, the `ElmQueryLetContext.letsContexts` was empty:
![Screen Shot 2023-03-30 at 1 27 05 PM](https://user-images.githubusercontent.com/56264529/228916807-a9e97818-b46d-4728-a387-51d081db9aee.png)

Not sure if this is a result of our setup of the processor or something internal. Please let me know if you need data dumps form live data.

CQL:
```
define "test":
    [MedicationRequest: "Androgen deprivation therapy for Urology Care"] ADTActive
     where ADTActive.status in { 'active', 'completed' }
        and ADTActive.intent in { 'order', 'original-order', 'reflex-order', 'filler-order', 'instance-order' }
        and ADTActive.doNotPerform is not true
     return QICoreCommon."Earliest"(
       "NormalizePeriod"(
         First(
            (ADTActive.dosageInstruction DoseInstruction
                let DoseTiming: DoseInstruction.timing,
                    FirstDoseEventInTiming: First(DoseTiming.event DoseEvent return DoseEvent sort ascending)
                return FirstDoseEventInTiming 
              ) DoseTimingEvent
                return DoseTimingEvent
                sort ascending
            ), 
         First(
            ( collapse (ADTActive.dosageInstruction.timing.repeat.bounds DoseTime
                return DoseTime.toInterval()) 
                ) DrugPeriod
                sort by start of $this
            )
         )
      )

define function "NormalizePeriod"(pointInTime DateTime, dateTimeInterval Interval<DateTime>): 
  if pointInTime is not null then Interval[pointInTime, pointInTime] 
  else if dateTimeInterval is not null then dateTimeInterval
  else null as Interval<DateTime>
```

**Comment by jenseeman** [comment]
Related to- https://oncprojectracking.healthit.gov/support/browse/BONNIEMAT-1415

**Comment by brynrhodes** [comment]
Included in 2.11.0
